spring-integration-core/src/main/java/org/springframework/integration/util/FunctionIterator.java

```diff
@@ -61,8 +102,23 @@ public V next() {
 
 	@Override
 	public void close() throws IOException {
-		if (this.iterator instanceof Closeable) {
-			((Closeable) this.iterator).close();
+		if (this.iterator instanceof AutoCloseable) {
+			try {
+				((AutoCloseable) this.iterator).close();
+			}
+			catch (Exception e) {
+				// NOSONAR
+			}
+		}
+		if (!this.root.equals(this.iterator)) {
+			if (this.root instanceof AutoCloseable) {
+				try {
+					((AutoCloseable) this.root).close();
+				}
+				catch (Exception e) {
+					// NOSONAR
```

Comment:
I believe we can change a `CloseableIterator` to the `AutoCloseable` now as well...